src/test/groovy/com/github/jk1/license/filter/LicenseBundleNormalizerFuncSpec.groovy

```diff
@@ -0,0 +1,272 @@
+                forTesting "io.netty:netty-common:4.1.17.Final" // license-name: "Apache License, Version 2.0"
+            }
+        """
+        normalizerFile << """
+              "transformationRules" : [
+                { "bundleName" : "apache2", "licenseNamePattern" : ".*The Apache Software License, Version 2.0.*" }
+              ]
+            }
+        """
+
+        when:
+        def runResult = GradleRunner.create()
+                .withProjectDir(testProjectDir.root)
+                .withArguments('generateLicenseReport')
+                .withPluginClasspath()
+                .build()
+
+        def result = jsonSlurper.parse(licenseResultJsonFile)
+        println(licenseResultJsonFile.text)
+
+        then:
+        runResult.task(":generateLicenseReport").outcome == TaskOutcome.SUCCESS
+
+        result.dependencies.size() == 2
+        result.dependencies*.moduleLicenses.flatten()*.moduleLicense.toSet() == ["Apache License, Version 2.0", null].toSet()
+        result.dependencies*.moduleLicenses.flatten()*.moduleLicenseUrl.toSet() == ["http://www.apache.org/licenses/LICENSE-2.0"].toSet()
+    }
+
+    def "normalizes dependencies by configured license url"() {
+        buildFile << """
+            dependencies {
+                forTesting "org.jetbrains:annotations:13.0"     // license-url: "http://www.apache.org/licenses/LICENSE-2.0.txt"
+                forTesting "io.netty:netty-common:4.1.17.Final" // license-url: "http://www.apache.org/licenses/LICENSE-2.0"
+            }
+        """
+        normalizerFile << """
+              "transformationRules" : [
+                { "bundleName" : "apache2", "licenseUrlPattern" : ".*http://www.apache.org/licenses/LICENSE-2.0.txt.*" }
+              ]
+            }
+        """
+
+        when:
+        def runResult = GradleRunner.create()
+                .withProjectDir(testProjectDir.root)
+                .withArguments('generateLicenseReport')
+                .withPluginClasspath()
+                .build()
+
+        def result = jsonSlurper.parse(licenseResultJsonFile)
+
+        then:
+        runResult.task(":generateLicenseReport").outcome == TaskOutcome.SUCCESS
+
+        result.dependencies.size() == 2
+        result.dependencies*.moduleLicenses.flatten()*.moduleLicense.toSet() == ["Apache License, Version 2.0", null].toSet()
+        result.dependencies*.moduleLicenses.flatten()*.moduleLicenseUrl.toSet() == ["http://www.apache.org/licenses/LICENSE-2.0"].toSet()
+    }
+
+    def "normalizes manifest with pom data"() {
+        buildFile << """
+            dependencies {
+                forTesting "joda-time:joda-time:2.9.7" // has manifest and pom information
+            }
+        """
+        normalizerFile << """
+              "transformationRules" : [
+                   { "bundleName" : "apache2", "licenseNamePattern" : "Apache 2.*" }
+              ]
+            }
+        """
+
+        when:
+        def runResult = GradleRunner.create()
```

Comment:
The `when` stage seems to be the same in all these tests - factor out a helper method? You could potentially also factor out a helper method for the `then` stage, which is very similar for most of the tests.